launcher/src/main/java/org/wildfly/core/launcher/JavaVersionComparator.java

```diff
@@ -59,40 +57,20 @@ public static int compareVersion(final String version1, final String version2) {
 
     @Override
     public int compare(final String o1, final String o2) {
-        final String[] vs1 = o1.split("\\.");
-        final String[] vs2 = o2.split("\\.");
-        final int len = (vs1.length > vs2.length ? vs1.length : vs2.length);
-        final List<Integer> v1 = convert(vs1, len);
-        final List<Integer> v2 = convert(vs2, len);
-        int result = 0;
-        for (int i = 0; i < len; i++) {
-            final Integer vi1 = v1.get(i);
-            final Integer vi2 = v2.get(i);
-            result = vi1.compareTo(vi2);
-            if (result != 0) {
-                break;
-            }
-        }
-        return result;
+        int version1 = getMajorVersion(o1);
+        int version2 = getMajorVersion(o2);
+        return Integer.compare(version1, version2);
     }
 
-    private static List<Integer> convert(final String[] version, final int len) {
-        final List<Integer> result = new ArrayList<>(len);
-        for (int i = 0; i < len; i++) {
-            if (i < version.length) {
-                final String s = version[i];
-                if ("x".equalsIgnoreCase(s)) {
-                    result.add(0);
-                } else {
-                    try {
-                        result.add(Integer.valueOf(s));
-                    } catch (NumberFormatException e) {
-                        throw new IllegalArgumentException(String.format("Version part %s is not a valid integer", s), e);
-                    }
-                }
-            } else {
-                result.add(0);
-            }
+    private int getMajorVersion(String versionString) {
+        final String[] vs = versionString.split("\\.");
```

Comment:
I just kept it as it was in previous version of code. 
Makes sense to simplify it as much as possible.